### ISSUE TITLE
[Bug] Updating the `layers` prop each frame in non-interleaved `MapboxOverlay` leads to increased opacity and flickering

### ISSUE THREAD

**Issue by JayFoxRox** [visgl/deck.gl#8529]
### Description

When zooming or moving the deck.gl layers are rendered more than once, leading to increased opacity or flickering.

### Flavors

- [ ] Script tag
- [ ] React
- [ ] Python/Jupyter notebook
- [X] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

Should not flicker or change opacity.

### Steps to Reproduce

Controlled by `enableBug3` in https://github.com/JayFoxRox/deck.gl-bugs/blob/master/src/DeckGlBugs.tsx

### Environment

- Framework version: deck.gl@9.0.0-beta.4
- Browser: All
- OS: Probably all, but at least macOS (Not sure if relevant, but this is on a 120Hz display)

**Comment by Pessimistress** [comment]
This is new in v9. @donmccurdy @ibgreen looks like calling `deck.redraw()` doesn't clear the canvas?

**Comment by donmccurdy** [comment]
Yes, and agreed — probably redraw() should clear the canvas by default, unless otherwise specified? Opened a PR:

- https://github.com/visgl/deck.gl/pull/8543

**Comment by felixpalmer** [comment]
@donmccurdy great to see this fixed, now that I think about it I recall seeing flickering behavior in other places, like 3D tiles with an overlay
